File: authorization-service/src/main/java/com/mazeltov/authorization/service/dao/repository/UserRepository.java
package com.mazeltov.authorization.service.dao.repository;

import org.springframework.data.jpa.repository.*;
import org.springframework.security.core.userdetails.*;
import org.springframework.stereotype.*;

import java.util.*;

@Repository
public interface UserRepository extends JpaRepository<com.mazeltov.authorization.service.dao.model.User, Long> {
    Optional<com.mazeltov.authorization.service.dao.model.User> findByUsername(String username);
}

File: recommendation-service/src/main/kotlin/com/mazeltov/recommendationservice/dao/recommendation/repository/UserRepository.kt
package com.mazeltov.recommendationservice.dao.recommendation.repository

import com.mazeltov.recommendationservice.dao.recommendation.model.*
import org.springframework.data.jpa.repository.*
import org.springframework.stereotype.*
import java.util.*

@Repository
interface UserRepository : JpaRepository<User, Long> {

    fun findByUserId(userId: Long): Optional<User>
}
